packages/SwingSet/test/upgrade/bootstrap-scripted-upgrade.js

```diff
@@ -282,5 +282,26 @@ export const buildRootObject = () => {
 
       return [paramA, paramB];
     },
+
+    buildV1WithVatParameters: async () => {
+      const bcap1 = await E(vatAdmin).getNamedBundleCap('ulrik1');
+      const vp1 = { number: 1, marker };
+      const options1 = { vatParameters: vp1 };
+      const res = await E(vatAdmin).createVat(bcap1, options1);
+      ulrikAdmin = res.adminNode;
+      ulrikRoot = res.root;
+      const param1 = await E(ulrikRoot).getParameters();
+      return param1;
+    },
+
+    upgradeV2WithVatParameters: async () => {
+      const bcap2 = await E(vatAdmin).getNamedBundleCap('ulrik2');
+      const vp2 = { number: 2, marker };
```

Comment:
It could, but I was relying upon the number to distinguish the original vs the new. I wanted the marker there to test that 1: krefs work, 2: refcounts are managed correctly. But I'm not checking refcounts in this test.. I do some of that in `create-vat.test.js`, but maybe I should add some here too.

If I was actually implementing the "vP-lacking `upgrade()` re-uses the retained vP" feature here, then this test would need to be bigger: use three markers, m1+m2 in the original, upgrade from v1 to v2 with no `options.vatParameters`, then upgrade from v2 to v3 with (m1+m3) in the vP. And we'd check that the refcount remains the same across the first upgrade (still retained, still received by the vat), but the second upgrade drops m2 and holds a refcount on m3.

But for this "merely retain them" change, that isn't necessary.

Reply:
I added a clause to check the refcounts in this test. It's muddled, because I'm not forcing BOYD, and the `kpResolution()` calls add extra ones, and `device-vat-admin` doesn't do GC, so the count is 6 instead of 2, and the second call sees a count of 7 because of the extra `kpResolution()` pin. But I think it still does the job of making sure that the saved parameters get a refcount,